aidial_adapter_openai/utils/tokenizer.py

```diff
@@ -30,11 +38,39 @@ def __init__(self, model: str) -> None:
                 "or declare it as a model which doesn't require tokenization through tiktoken.",
             ) from e
 
-    def calculate_text_tokens(self, text: str) -> int:
+    def tokenize_text(self, text: str) -> int:
         return len(self.encoding.encode(text))
 
+    def tokenize_response(self, resp: ChatCompletionResponse) -> int:
+        return sum(map(self._tokenize_response_message, resp.messages))
+
+    def _tokenize_object(self, obj: Any) -> int:
```

Comment:
I like the idea of JSON type, but it will lead to **a lot** of additional type checks, the code will be become unreadable.

We should rather migrate to the type provided by the `openai` library, so we don't need to deal with the raw JSON in the first place.